topics/imaging/tutorials/astronomy-source-extractor/tutorial.md

```diff
@@ -0,0 +1,226 @@
+-  [SEP paper](https://joss.theoj.org/papers/10.21105/joss.00058)
+-  [Source Extractor for Dummies](https://arxiv.org/abs/astro-ph/0512139)
+-  [Source Extractor paper](https://ui.adsabs.harvard.edu/abs/1996A%26AS..117..393B/abstract)
+-  [Source Extractor website](https://www.astromatic.net/software/sextractor/)
+
+> <agenda-title></agenda-title>
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+
+## Background Estimation and Thresholding
+
+Before source detection, the tool estimates the image background. This is done by dividing the image into a grid of boxes, each with a default size of:
+``` python
+bw = 64  # box width in pixels
+bh = 64  # box height in pixels
+```
+Within each box, the pixel histogram is clipped to remove outliers, and the background level is estimated using a mode approximation based on the median and mean of the clipped pixel values. 
+
+After background estimation, the tool identifies groups of pixels that exceed a defined brightness threshold.
+
+Detection Criteria:
+
+- Minimum Area: The number of connected pixels required to consider something a source.
+
+``` python
+minarea = 5 # default
+```
+
+- Threshold: Flux at pixel (j, i) must exceed:
+
+``` python
+thresh * err[j,i]
+```
+
+where:
+
+``` python
+thresh = 1.5 # default
+```
+
+The interpretation of ```err[j,i]``` depends on the ```err_option``` parameter:
+``` python
+err_option = 'float_globalrms'  # Use global RMS of the background (default)
+err_option = 'array_rms'        # Use a pixel-wise RMS array of the background
+err_option = 'none'             # Use 'thresh' as an absolute threshold
+```
+
+## Data Requirements 
+
+The source-extractor tool operates on a single image file, typically for astronomy, a sky image containing luminous sources. Optionally, users can supply a mask and/or a filter. 
+
+**Image:** 
+- Preferrably: light sources on a dark background.
+- Format: a single-channel 2D array stored as ```.tiff``` or ```.fits```. 
+
+**Mask (Optional):** 
+- Masks regions affected by bright sources (e.g. stars) to improve background estimation. 
+- Pixels with
+``` python
+value > maskthresh
+```
+or boolean ```True``` are masked.
+- Format: a single-channel 2D array stored as ```.tiff``` or ```.fits```.  
+
+**Filter Kernel (Optional):** 
+The filter kernel is used to smooth the input image, which can enhance the detection of faint and extended sources. However, in crowded fields, filtering may reduce performance by blending nearby objects.
+
+- If ```Filter Case``` is set to ```none```, no filtering is applied.
+- If ```Filter Case``` is ```default```, a built-in smoothing kernel is used:
+```markdown
+1 2 1
+2 4 2
+1 2 1
+```
+- If ```Filter Case``` is ```file```, you must provide a custom 2D array stored as plain text ```.txt``` file. The file should contain whitespace-separated values and must be readable with:
+```python
+import numpy as np
+kernel = np.loadtxt("filter.txt")
+```
```

Comment:
so up to this point one needs to do the preprocessing locally? And then move to Galaxy?

Reply:
Still not clear to me where this happens